*** howling_suppression.c.orig	2014-07-21 17:02:49.693284490 +0400
--- howling_suppression.c	2014-07-21 17:05:35.955717102 +0400
***************
*** 203,208 ****
--- 203,209 ----
  
  void Hs_Process(HsHandle *inst, const short *input, short *output)
  {
+     /*
      int freqCount = Hs_AnalyzeHowling(inst, input);
      if (!freqCount)
      {
***************
*** 220,228 ****
      for (short i = 0; i < freqCount; ++i)
          Hs_BiquadCalc(inst->filter + i,
                        inst->howlingFreq[i] / 100.0 / inst->blockLen);
  
      // Apply IIR filter
      const short *input_p = input;
      while (input_p < input + inst->blockLen)
!         *output++ = Hs_BiquadCascade(inst->filter, freqCount, *input_p++);
  }
--- 221,234 ----
      for (short i = 0; i < freqCount; ++i)
          Hs_BiquadCalc(inst->filter + i,
                        inst->howlingFreq[i] / 100.0 / inst->blockLen);
+ */
+     // PARAMS
+     inst->filter[0].peakGain = -5;  // dB
+     int freq                 = 700; // Hz
  
      // Apply IIR filter
+     Hs_BiquadCalc(inst->filter, freq / 8000.0);
      const short *input_p = input;
      while (input_p < input + inst->blockLen)
!         *output++ = Hs_BiquadCascade(inst->filter, 1, *input_p++);
  }
